pkg/resource/deploy/source_eval.go

```diff
@@ -470,6 +470,7 @@ func (d *defaultProviders) getDefaultProviderRef(req providers.ProviderRequest)
 // resmon implements the pulumirpc.ResourceMonitor interface and acts as the gateway between a language runtime's
 // evaluation of a program and the internal resource planning and deployment logic.
 type resmon struct {
+	ctx                       *plugin.Context                    // the plugin context
```

Comment:
Dunno if you need Context for later changes but if not maybe just pass the diag.Sink object instead.